src/graph/graph.c

```diff
@@ -554,24 +574,73 @@ int Graph_ConnectNodes(Graph *g, NodeID src, NodeID dest, int r, Edge *e) {
 	GrB_Matrix_setElement_BOOL(tadj, true, dest, src);
 	GrB_Index I = src;
 	GrB_Index J = dest;
-	id = SET_MSB(id);
+	edgeId = SET_MSB(edgeId);
 	info = GxB_Matrix_subassign_UINT64   // C(I,J)<Mask> = accum (C(I,J),x)
 		   (
 			   relationMat,         // input/output matrix for results
 			   GrB_NULL,            // optional mask for C(I,J), unused if NULL
 			   _graph_edge_accum,   // optional accum for Z=accum(C(I,J),x)
-			   id,                  // scalar to assign to C(I,J)
+			   edgeId,              // scalar to assign to C(I,J)
 			   &I,                  // row indices
 			   1,                   // number of row indices
 			   &J,                  // column indices
 			   1,                   // number of column indices
 			   GrB_NULL             // descriptor for C(I,J) and Mask
 		   );
 	assert(info == GrB_SUCCESS);
+}
+
+int Graph_ConnectNodes(Graph *g, NodeID src, NodeID dest, int r, Edge *e) {
+
+	Node srcNode;
+	Node destNode;
+
+	assert(Graph_GetNode(g, src, &srcNode));
+	assert(Graph_GetNode(g, dest, &destNode));
```

Comment:
I know you didn't add this, but it seems like a lot of extra work just to assert. I propose deleting this - thoughts?

Reply:
Isn't it part of the graph API for creating new edges during queries?